### ISSUE TITLE
Clean up temp indexes

### ISSUE THREAD

**Issue by adamnovak** [vgteam/vg#684]
VG creates a temp FASTA file during unit testing, and ends up indexing it. I think we clean up the base file, but we don't clean up the FASTA index generated based on its name, and we should.

I think we also might have a slight security problem, because someone could symlink the index name between when we check for the index's existence and when we fill it with data, and clobber files they couldn't normally touch by getting us to write FASTA index data over them.

Both these problems could be soled by using a temp directory rather than a temp file.

**Comment by edawson** [comment]
Slightly related - is it possible to easily check if indexes are older than the file they index? I've been bit by old indices causing uninformative vg errors more times than I can count.

**Comment by adamnovak** [comment]
We might want to try and push that down into fastahack.

**Comment by ekg** [comment]
Agreed

**Comment by edawson** [comment]
I also agree but don't know how to implement it - if y'all give me some direction I'm happy to fix.
